vms/evm/sync/customrawdb/accessors_metadata.go

```diff
@@ -0,0 +1,173 @@
+func WriteAcceptorTip(db ethdb.KeyValueWriter, hash common.Hash) error {
+	return db.Put(acceptorTipKey, hash[:])
+}
+
+// ReadAcceptorTip reads the hash of the last accepted block that was fully processed.
+// If there is no value present (the index is being initialized for the first time), then the
+// empty hash is returned.
+func ReadAcceptorTip(db ethdb.KeyValueReader) (common.Hash, error) {
+	ok, err := db.Has(acceptorTipKey)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	if !ok {
+		return common.Hash{}, ErrEntryNotFound
+	}
+	h, err := db.Get(acceptorTipKey)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	if len(h) != common.HashLength {
+		return common.Hash{}, fmt.Errorf("%w: length %d", ErrInvalidData, len(h))
+	}
+	return common.BytesToHash(h), nil
+}
+
+// ReadChainConfig retrieves the consensus settings based on the given genesis hash.
+// The provided `upgradeConfig` (any JSON-unmarshalable type) will be populated if present on disk.
+func ReadChainConfig[T any](db ethdb.KeyValueReader, hash common.Hash, upgradeConfig *T) (*params.ChainConfig, error) {
+	config := rawdb.ReadChainConfig(db, hash)
+	if config == nil {
+		return nil, ErrEntryNotFound
+	}
+
+	upgrade, _ := db.Get(upgradeConfigKey(hash))
+	if len(upgrade) == 0 {
+		return config, nil
+	}
+
+	if err := json.Unmarshal(upgrade, upgradeConfig); err != nil {
+		return nil, ErrInvalidData
+	}
+
+	return config, nil
+}
+
+// WriteChainConfig writes the chain config settings to the database.
+// The provided `upgradeConfig` (any JSON-marshalable type) will be stored alongside the chain config.
+func WriteChainConfig[T any](db ethdb.KeyValueWriter, hash common.Hash, config *params.ChainConfig, upgradeConfig T) error {
+	rawdb.WriteChainConfig(db, hash, config)
+	if config == nil {
+		return nil
+	}
+
+	data, err := json.Marshal(upgradeConfig)
+	if err != nil {
+		return err
+	}
+	if err := db.Put(upgradeConfigKey(hash), data); err != nil {
+		return err
+	}
+	return nil
+}
+
+// writeCurrentTimeMarker writes a marker of the current time in the db at `key`.
+func writeCurrentTimeMarker(db ethdb.KeyValueStore, key []byte) error {
+	data, err := rlp.EncodeToBytes(uint64(time.Now().Unix()))
+	if err != nil {
+		return err
+	}
+	return db.Put(key, data)
+}
+
+// readTimeMarker reads the timestamp stored at `key`
+func readTimeMarker(db ethdb.KeyValueStore, key []byte) (time.Time, error) {
+	// Check existence first to map missing marker to a stable sentinel error.
+	ok, err := db.Has(key)
+	if err != nil {
+		return time.Time{}, err
+	}
+	if !ok {
+		return time.Time{}, ErrEntryNotFound
+	}
+
+	data, err := db.Get(key)
+	if err != nil {
+		return time.Time{}, err
+	}
+	if len(data) == 0 {
+		return time.Time{}, ErrEntryNotFound
+	}
+
+	var unix uint64
+	if err := rlp.DecodeBytes(data, &unix); err != nil {
+		return time.Time{}, fmt.Errorf("%w: %w", ErrInvalidData, err)
+	}
+
+	return time.Unix(int64(unix), 0), nil
+}
+
+// upgradeConfigKey = upgradeConfigPrefix + hash
```

Comment:
nit: Some of these in-line comments are documenting self-documenting code - could we remove these? Comments have a maintenance cost associated with them since they need to be kept in-sync w/ code, so if code is reasonably self-documenting we should generally avoid adding comments to them.